src/AppBundle/Controller/MembershipController.php

```diff
@@ -34,18 +37,68 @@ public function registerAction(Request $request): Response
             $this->get('app.membership_request_handler')->handle($membership);
             $this->addFlash('info', $this->get('translator')->trans('adherent.registration.success'));
 
-            return $this->redirectToRoute('app_membership_register');
+            if ($membership->hasAdherent()) {
+                $this->get('app.membership_utils')->createRegisteringDonation($membership->getAdherent());
+            }
+
+            return $this->redirectToRoute('app_membership_donate');
         }
 
-        return $this->render('membership/registration.html.twig', [
+        return $this->render('membership/register.html.twig', [
             'membership' => $membership,
             'form' => $form->createView(),
             'countries' => UnitedNationsBundle::getCountries($request->getLocale()),
         ]);
     }
 
     /**
-     * This action enables a guest user to activate his\her newly created
+     * This action enables a new user to donate as a second step of the
+     * registration process, thus he/she may not be logged-in/activated yet.
+     *
+     * @Route("/inscription/don", name="app_membership_donate")
+     * @Method("GET|POST")
+     */
+    public function donateAction(Request $request): Response
+    {
+        $memberShipUtils = $this->get('app.membership_utils');
+
+        if (!$donation = $memberShipUtils->getRegisteringDonation()) {
+            throw $this->createNotFoundException('The adherent has not been successfully redirected from the registration page.');
+        }
+
+        $form = $this->createForm(DonationType::class, $donation, [
+            'locale' => $request->getLocale(),
+            'submit_label' => 'adherent.submit_donation_label',
+            'sponsor_form' => false,
+        ])
+            // TODO add this field only if anonymous (adherent not activated yet)
```

Comment:
I don't think so, we don't know yet who will fix it and when this will be fixed, but we know it needs to be :)